MarkupGatewayPlugin.inc.php

```diff
@@ -257,25 +257,98 @@ function fetch($args, $request) {
 	 */
 	protected function _buildMetadata($journal, $submission)
 	{
-		$locale = AppLocale::getLocale();
-
+		$locale = ($submission->getLanguage() != '') ? $submission->getLanguage() : $journal->getPrimaryLocale();
+		
+		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+		$issueDao = DAORegistry::getDAO('IssueDAO');		
+		$sectionDao = DAORegistry::getDAO('SectionDAO');
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+		
+		/* Authors */
 		$authors = array_map(function($author)
 		{
 			return array (
-					'name' => $author->getFullName(),
+					'firstName' => $author->getFirstName(),
+					'lastName' => $author->getLastName(),
 					'email' => $author->getEmail(),
+					'orcid' => $author->getOrcid(),
+					'affiliation' => $author->getLocalizedAffiliation(),
+					'country' => $author->getCountry(),
+					'bio' => $author->getLocalizedBiography(),
+					'contribType' => $userGroupDao->getById($author->getUserGroupId()),
```

Comment:
this is throwing:

`[Tue Aug 08 16:45:10.472925 2017] [:error] [pid 11966] [client 142.58.129.81:55038] PHP Notice:  Undefined variable: userGroupDao in /var/www/html/ojs-plugin-alpha-12/plugins/generic/markup/MarkupGatewayPlugin.inc.php on line 278`
`[Tue Aug 08 16:45:10.472962 2017] [:error] [pid 11966] [client 142.58.129.81:55038] PHP Fatal error:  Call to a member function getById() on null in /var/www/html/ojs-plugin-alpha-12/plugins/generic/markup/MarkupGatewayPlugin.inc.php on line 278`

it's possible that you're more familiar with this part of OJS than I am, unless @asmecher wants to weigh in?

Reply:
The usergroupdao issue was related to the array_map function, I changed that to an foreach loop instead. Should work now.